dp-cluster-setup-utility.py

```diff
@@ -895,6 +862,154 @@ def streamsmsgmgr_url(self):
   def host_name(self, service_name, component_name):
     return self.ambari.cluster.service(service_name).component(component_name).host_names()[0]
 
+
+class DpProxyTopology:
+  TEMPLATE = """<?xml version="1.0" encoding="UTF-8"?>
+  <topology>
+     <uri>{knox_url}/gateway/dp-proxy</uri>
+     <name>dp-proxy</name>
+     <timestamp>{timestamp}</timestamp>
+     <generated>false</generated>
+     <gateway>
+        <provider>
+           <role>federation</role>
+           <name>SSOCookieProvider</name>
+           <enabled>true</enabled>
+           <param>
+              <name>sso.authentication.provider.url</name>
+              <value>{knox_url}/gateway/knoxsso/api/v1/websso</value>
+           </param>
+        </provider>
+        <provider>
+           <role>identity-assertion</role>
+           <name>Default</name>
+           <enabled>true</enabled>
+        </provider>
+     </gateway>
+     <service>
+        <role>AMBARI</role>
+        <version>0.2.2.0</version>
+        <url>{ambari_protocol}://{ambari_host}:{ambari_port}</url>
+     </service>
+     {ranger}
+     {atlas_api}
+     {dpprofiler}
+     {beacon}
+     {streamsmsgmgr}
+  </topology>"""
+
+  def __init__(self, ambari, role_names, topology_util, name='dp-proxy'):
```

Comment:
These menthods __init__ and deploy are common and same can be called for both dp-proxy and beacon-proxy topologies